chain/chain/src/migrations.rs

```diff
@@ -0,0 +1,82 @@
+use crate::store::ChainStoreAccess;
+use crate::types::RuntimeAdapter;
+use near_chain_primitives::error::Error;
+use near_primitives::checked_feature;
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::{EpochId, ShardId};
+
+/// Get epoch id of the last block in which chunk for the given shard was presented.
+fn get_epoch_id_of_last_block_with_chunk(
```

Comment:
What's the reason for implementing these functions as standalone functions rather than methods of `ChainUpdate` or `Chain`?

Reply:
We have calls of `runtime.apply_transactions` in many places, like `apply_block_at_height` in `neard/migrations`, `state-viewer`. Also there is `validate_chunk_state_challenge`, where we don't call it yet. 
In all these places we currently create only `ChainStore` or just block headers.